api/core.py

```diff
@@ -416,7 +430,11 @@ def _archivestream(self, ticket):
         stream = cStringIO.StringIO()
         with tarfile.open(mode='w|', fileobj=stream) as archive:
             for filepath, arcpath, _ in ticket['target']:
-                yield archive.gettarinfo(filepath, arcpath).tobuf()
+                try:
+                    yield archive.gettarinfo(filepath, arcpath).tobuf()
+                except Exception as e:
```

Comment:
You're right. This would only occur if something bad has happened. We're likely to have a few bugs like #115, though, so this should remain until we are much more confident in our codebase. There is also a race condition between uploads and downloads. We should think about the order of operations for uploads. Proposal: deposit file first, then create DB entries. We're already planning a cron job that looks for non-referenced files.